server/handler/details.go

```diff
@@ -121,31 +120,20 @@ func (h *Details) ServeHTTP(w http.ResponseWriter, r *http.Request) error {
 
 	config, err := h.ConfigFetcher.ConfigForPR(ctx, prctx, client)
 	data.PolicyURL = getPolicyURL(pr, config)
-
 	if err != nil {
-		data.Error = errors.WithMessage(err, fmt.Sprintf("Failed to fetch configuration at ref=%s", config.Ref))
+		data.Error = errors.WithMessage(err, fmt.Sprintf("Failed to fetch configuration: %s", config))
 		return h.render(w, data)
 	}
 
-	if config.Missing() {
-		data.Error = errors.New(config.Description())
+	evaluator, err := h.Base.ValidateFetchedConfig(ctx, prctx, client, config, common.TriggerAll)
+	if evaluator == nil {
+		data.Error = err
 		return h.render(w, data)
 	}
 
-	if config.Invalid() {
-		data.Error = errors.WithMessage(config.Error, config.Description())
-		return h.render(w, data)
-	}
-
-	evaluator, err := policy.ParsePolicy(config.Config)
-	if err != nil {
-		data.Error = errors.WithMessage(err, fmt.Sprintf("invalid policy at ref \"%s\"", config.Ref))
-		return h.render(w, data)
-	}
-
-	result := evaluator.Evaluate(ctx, prctx)
+	result, err := h.Base.EvaluateFetchedConfig(ctx, prctx, client, evaluator, config)
```

Comment:
Given the comment above about review requests, I think we should look at inlining `EvaluateFetchedConfig` here and accept the duplication so we can more easily drop parts we don't want. The introduction of `ValidateFetchedConfig` helps remove some of the existing duplication, so maybe that's enough.

Reply:
I moved the success case status update logic back into `EvaluateFetchedConfig`, and it feels more like a self contained block of logic now. Errors are thrown when evaluation produces an error or validation of the error status has a problem. 

The callers of `EvaluateFetchedConfig` handle the err/result in their own ways, but don't seem to need more customization of what actually occurs in this method. Therefore, I think we can probably keep it as it's own method. Does that sound ok?